packages/react-reconciler/src/ReactFiberContext.js

```diff
@@ -241,6 +252,7 @@ function pushContextProvider(workInProgress: Fiber): boolean {
 
```

Comment:
Your commit states that you removed some calls to `isContextProvider` but you never actually removed it from `pushContextProvider`.

Reply:
> I removed some of them in the previous commit, and deleted a few more in this one. I left a few behind because the remaining ones would require additional refactoring that feels outside the scope of this PR.

Reply:
I'm not a fan of the double checking. I don't understand where this is going. Is push/pop going to assume that whether it is a provider or not has been asserted or not? Might be worth while doing in this.

Reply:
Ok I'll get rid of it

Reply:
Although I think passing in the Component might be better than checking it on the outside. Every time we check things on classes, or promises, is a deopt since they don't have consistent hidden classes. We should try to minimize the effect of deopts by moving them to a shared function like what we had.

Reply:
Right but the shared function should accept the resolved component type, not the whole fiber. Unless I misunderstand.